src/server/storage.zig

```diff
@@ -252,21 +252,43 @@ pub fn loadRegionFileAndIncreaseRefCount(wx: i32, wy: i32, wz: i32, voxelSize: u
 }
 
 pub const ChunkCompression = struct { // MARK: ChunkCompression
-	const CompressionAlgo = enum(u32) {
-		deflate_with_position = 0,
-		deflate = 1,
+	const ChunkCompressionAlgo = enum(u32) {
+		deflate_with_position_no_block_entities = 0,
+		deflate_no_block_entities = 1,
 		uniform = 2,
-		deflate_with_8bit_palette = 3,
-		_,
+		deflate_with_8bit_palette_no_block_entities = 3,
+		deflate = 4,
+		deflate_with_8bit_palette = 5,
+	};
+	const BlockEntityCompressionAlgo = enum(u8) {
+		raw = 0, // TODO: Maybe we need some basic compression at some point. For now this is good enough though.
 	};
-	pub fn compressChunk(allocator: main.heap.NeverFailingAllocator, ch: *chunk.Chunk, allowLossy: bool) []const u8 {
-		if(ch.data.paletteLength == 1) {
-			var writer = BinaryWriter.initCapacity(allocator, @sizeOf(CompressionAlgo) + @sizeOf(u32));
 
-			writer.writeEnum(CompressionAlgo, .uniform);
-			writer.writeInt(u32, ch.data.palette[0].toInt());
+	const Target = enum {toClient, toDisk};
+
+	pub fn storeChunk(allocator: main.heap.NeverFailingAllocator, ch: *chunk.Chunk, comptime target: Target, allowLossy: bool) []const u8 {
+		var writer = BinaryWriter.init(allocator);
+
+		compressBlockData(ch, allowLossy, &writer);
+		compressBlockEntityData(ch, target, &writer);
+
+		return writer.data.toOwnedSlice();
+	}
+
+	pub fn loadChunk(ch: *chunk.Chunk, comptime side: main.utils.Side, data: []const u8) !void {
+		var reader = BinaryReader.init(data);
+		try decompressBlockData(ch, &reader);
+		decompressBlockEntityData(ch, side, &reader) catch |err| {
```

Comment:
Why do you handle them differently?

Reply:
Because failing to decompress block data is non-recoverable and will only happen if there is a data corruption.
Whereas we can get corrupted block entity data when someone just changes the block entity of an existing block, which should not discard the entire chunk.

Reply:
Oh right, actually the use case I mentioned is already handled in the function itself.
This here is just leftover debug code.